spirekey/tests/utils/merkle.test.ts

```diff
@@ -13,68 +13,163 @@ describe('Merkle', () => {
       });
     });
     describe('And we have 3 entries', () => {
-      it('should get the hash based on `H(H(A,B),H(C,0))`', () => {
+      it('should get the hash based on `H(H(A,B),C)`', () => {
         const hashes = ['A', 'B', 'C'];
-        expect(getRootHash(hashes)).toEqual('H(H(A,B),H(C,0))');
+        expect(getRootHash(hashes)).toEqual('H(H(A,B),C)');
       });
     });
     describe('And we have 5 entries', () => {
-      it('should get the hash based on `H(H(H(A,B),H(C,D)),H(H(E,0),H(0,0)))`', () => {
+      it('should get the hash based on `H(H(H(A,B),H(C,D)),E)`', () => {
         const hashes = ['A', 'B', 'C', 'D', 'E'];
-        expect(getRootHash(hashes)).toEqual(
-          'H(H(H(A,B),H(C,D)),H(H(E,0),H(0,0)))',
-        );
+        expect(getRootHash(hashes)).toEqual('H(H(H(A,B),H(C,D)),E)');
```

Comment:
Ahh so now we don't need the 0 values anymore

Reply:
Yes and the shape of the tree could be asymmetric, which for batch signing is not relevant, but now it has the same implementation as Chainweb's merkle-log